pkg/cache/clusterqueue.go

```diff
@@ -237,42 +247,53 @@ func (c *clusterQueue) inactiveReason() (string, string) {
 		return "Terminating", "Can't admit new workloads; clusterQueue is terminating"
 	case pending:
 		reasons := make([]string, 0, 3)
+		messages := make([]string, 0, 3)
 		if c.isStopped {
 			reasons = append(reasons, "Stopped")
+			messages = append(messages, "Is stopped.")
 		}
-		if c.hasMissingFlavors {
+		if len(c.missingFlavors) > 0 {
 			reasons = append(reasons, "FlavorNotFound")
+			messages = append(messages, fmt.Sprintf("References missing ResourceFlavor %v.", c.missingFlavors))
 		}
-		if c.hasMissingOrInactiveAdmissionChecks {
-			reasons = append(reasons, "CheckNotFoundOrInactive")
+		if len(c.missingAdmissionChecks) > 0 {
+			reasons = append(reasons, "AdmissionCheckNotFound")
+			messages = append(messages, fmt.Sprintf("References missing AdmissionChecks %v.", c.missingAdmissionChecks))
 		}
-
-		if c.hasMultipleSingleInstanceControllersChecks {
-			reasons = append(reasons, "MultipleSingleInstanceControllerChecks")
+		if len(c.inactiveAdmissionChecks) > 0 {
+			reasons = append(reasons, "AdmissionCheckInactive")
+			messages = append(messages, fmt.Sprintf("References inactive AdmissionChecks %v.", c.inactiveAdmissionChecks))
+		}
+		if len(c.multipleSingleInstanceControllersChecks) > 0 {
+			reasons = append(reasons, "MultipleSingleInstanceControllerAdmissionChecks")
+			for _, controller := range utilmaps.SortedKeys(c.multipleSingleInstanceControllersChecks) {
+				messages = append(messages, fmt.Sprintf("Only one AdmissionChecks of %v can be reference for controller %q.", c.multipleSingleInstanceControllersChecks[controller], controller))
```

Comment:
```suggestion
				messages = append(messages, fmt.Sprintf("references multiple admission checks %v for controller %q.", c.multipleSingleInstanceControllersChecks[controller], controller))
```

Reply:
it doesn't sound grammatically correct: "... only one AdmissionChecks of [check2 check3] can be reference for controller "controller2"".

To make it grammatically correct we could have "only one AdmissionCheck of [check2 check3] can be referenced for controller "controller2"", but IIUC this is not what is meant by the rule.

Reply:
I think this is yet a reason to move forward with the condition-based mechanism: https://github.com/kubernetes-sigs/kueue/issues/3094. It is inflexible for set custom messages per controller type, and the generic messages are very hard to understand. 

Ideally, in case we would say "CQ is referencing multiple instances of the MultiKueue AdmissionCheck: <instance names>".

We can revisit the messages once we drop the conditions.

Reply:
updated